catalog/core/catalog-core-standardframework/src/main/java/ddf/catalog/cache/impl/FileSystemPersistenceProvider.java

```diff
@@ -215,7 +168,60 @@ public void clear() {
     }
   }
 
-  private File getMapStoreFile(String key) {
-    return new File(getMapStorePath() + key + SER);
+  Object loadFromPersistence(String key) {
+    File file = getMapStoreFilePath(key).toFile();
+    if (!file.exists()) {
+      return null;
+    }
+    InputStream inputStream = null;
+    try {
+      inputStream = new FileInputStream(Paths.get(persistencePath.toString(), key + SER).toFile());
+      InputStream buffer = new BufferedInputStream(inputStream);
+
+      try (ObjectInput input = new ObjectInputStream(buffer)) {
+        return input.readObject();
+      }
+    } catch (IOException e) {
+      LOGGER.info("Unable to read object.", e);
+    } catch (ClassNotFoundException e) {
+      LOGGER.info("Class for object being read from stream does not exist.", e);
+    } finally {
+      IOUtils.closeQuietly(inputStream);
+    }
+    return null;
+  }
+
+  private Path getMapStoreFilePath(String key) {
+    return Paths.get(persistencePath.toString(), key + SER);
+  }
+
+  private FilenameFilter getFilenameFilter() {
+    FilenameFilter filter =
+        new FilenameFilter() {
+          @Override
+          public boolean accept(File file, String name) {
+            return name.toLowerCase(Locale.getDefault()).endsWith(SER);
+          }
+        };
+    return filter;
+  }
+
+  /**
+   * Retrieve root directory of all persisted Hazelcast objects for this cache. The path is relative
+   * to containing bundle, i.e., DDF install directory, or based on the "storage" property passed
+   * into the ctor
+   *
+   * @return the path to root directory where serialized objects will be persisted
+   */
+  Path generatePersistencePath() {
+    if (properties.isEmpty() || properties.getProperty(STORAGE_ROOT_PROPERTY).isEmpty()) {
+      return Paths.get(new AbsolutePathResolver(defaultPersistencePath.toString()).getPath());
```

Comment:
I'm really tempted to just remove the `AbsolutePathResolver` stuff and just rely on the fact that blueprint will be passing in the correct path when this is created. Thoughts?

Reply:
Good question. If we do, we need to make sure that blueprint handles `${ddf.home}` and other replacement variables fine (it should).

Reply:
I thought it would, but for some reason I had to add a `PropertyResolver` in the code, must be missing something simple here. How does it normally resolve the properties?